text/###-v2-provider-interface.md

```diff
@@ -0,0 +1,402 @@
+# Improved Provider Interface
```

Comment:
You don't like the header ;)?

Reply:
Oops, I completely forgot about that. I based this from an old gist I wrote to share some ideas on this and forgot to add the header when I adapted it for this. Fixed in: https://github.com/Qiskit/rfcs/pull/2/commits/d854980f610eacb8c2fbcda2ff2cdf1be3464cbd